france_connect/clients.py

```diff
@@ -0,0 +1,169 @@
+    def generate_state(self) -> str:
+        """Generate a random state in hexadecimal string (64 characters)."""
+        return secrets.token_hex(48)
+
+    def get_authentication_url(
+        self, acr_values: List[ACRValues] = [ACRValues.EIDAS1.value], nonce: str = None, state: str = None
+    ) -> Tuple[str, str, str]:
+        """Return the authentication URL, nonce and state."""
+        nonce = nonce or self.generate_nonce()
+        state = state or self.generate_state()
+        parameters = (
+            f"client_id={self.client_id}"
+            f"&response_type=code"
+            f"&scope={'%20'.join(self.scopes)}"
+            f"&acr_values={' '.join(acr_values)}"
+            f"&nonce={nonce}"
+            f"&state={state}"
+            f"&redirect_uri={self.post_login_callback_url}"
+        )
+        url = f"{self.fc_base_url}{self.fc_authorization_url}?{parameters}"
+        return url, nonce, state
+
+    def get_configuration(self) -> dict:
+        """Return the OpenID configuration."""
+        response = requests.get(f"{self.fc_base_url}/api/v2/.well-known/openid-configuration", timeout=10)
+        if response.status_code != 200:
+            logger.error(f"FranceConnect - Failed to retrieve configuration: {response.status_code}")
+            return {}
+        return response.json()
+
+    def get_access_token(self, code: str) -> dict:
+        """Exchange the authorization code for an access token."""
+        token_url = f"{self.fc_base_url}{self.fc_token_url}"
+        data = {
+            "grant_type": "authorization_code",
+            "code": code,
+            "redirect_uri": self.post_login_callback_url,
+            "client_id": self.client_id,
+            "client_secret": self.client_secret,
+        }
+
+        response = requests.post(token_url, data=data, headers={"Accept": "application/json"}, timeout=10)
+
+        if response.status_code != 200:
+            logger.error(f"FranceConnect - Failed to retrieve access_token: {response.status_code}")
+            return {}
+        return response.json()
+
+    def _verify_jwt(self, token: str, token_type=str) -> dict:
+        """Verify the JWT signature and returns decoded token."""
+        jwks_client = PyJWKClient(f"{self.fc_base_url}{self.fc_jwks_url}")
+
+        try:
+            # Get the signing key from the JWT
+            key = jwks_client.get_signing_key_from_jwt(token)
+            # Decode and verify the JWT with the signing key
+            return jwt.decode(token, key.key, algorithms=["RS256"], audience=self.client_id)
+        except ExpiredSignatureError:
+            msg = f"FranceConnect - FranceConnect {token_type} has expired."
+            logger.error(msg)
+            return {JWTErrorCodes.EXPIRED_SIGNATURE.value: msg}
+        except InvalidTokenError as e:
+            msg = f"FranceConnect - FranceConnect invalid {token_type}: {e}"
+            logger.error(msg)
+            return {JWTErrorCodes.INVALID_TOKEN.value: msg}
```

Comment:
You should let these exceptions propagate and be handled by the code using your package.